LibTCPServer.emx

```diff
@@ -1050,152 +1077,152 @@ catch (Poco::Exception& exc) {
+		s.replace(pos, quote.size(), escaped);
 	
-	void run()
-	{
-		thread = Thread::current();
+	s.insert(0, quote);
+	s.insert(s.size(), quote);
+	return s;
+}
 
-		if (capsuleInst.config.logToStdOut)
-			std::cout << "TCP request from " << this->socket().peerAddress().toString() << std::endl;
-		try
-		{
-			char buffer[1024];
-			size_t n;
-			std::string received;
-			do {
-				n = socket().receiveBytes(buffer, sizeof(buffer));
-				if (n > 0)
-					received.append(buffer, n);
-			} 
-			while (n == sizeof(buffer) && socket().available() > 0);
-						
-			//cout << "Data read: " << received << endl << flush;
-
-			std::pair<std::string, LocalServerConnection*>* data = new std::pair<std::string, LocalServerConnection*>(received, this);
-			capsuleInst.external.dataPushBack(data);
-
-			if (capsuleInst.external.raise() == 0) {
-				// The capsule is not ready to receive the external event. Wait a little and try 2 more times before giving up.
-				thread->sleep(500);
-				if (capsuleInst.external.raise() == 0) {
-					thread->sleep(500);
-					if (capsuleInst.external.raise() == 0) {
-						replyStatus(Status::Error, "Failed to notify TCPServer capsule about received message after 3 attempts. Message is lost!");
-						// Need to remove the data object since this LocalServerConnection soon will become invalid						
-						capsuleInst.external.dataDelete(data);
-						return;								
-					}
-				}					
-			}
+static const std::string NL = "\n";
 
-#ifdef _DEBUG
-			long sleepDuration = 1000000;
-#else
-			long sleepDuration = capsuleInst.config.maxWaitForReply;
-#endif
-			// Keep the server connection open until a reply is received (or a timeout occurs).
-			if (thread->trySleep(sleepDuration)) {
-				// Timeout			
-				std::ostringstream s;
-				s << "No response from RT application within " << sleepDuration << " ms.";
-				replyStatus(Status::Error, s.str());
-			}			
-		}
-		catch (Poco::Exception& exc) {
-			replyStatus(Status::Error, "Failed to read from socket: " + exc.displayText());			
-		}
+class LocalServerConnection : public ITCPConnection {
+	TCPServer_Actor& server;
+	RTTcpSocket& listenSock;
+	RTTcpSocket clientSock;
+	RTIOMonitor ioMonitorClient;
+	long maxWaitForReply;
+	bool logToStdOut;
+	std::atomic_bool hasReplied;
+
+	friend class ServerJob;
+
+public:
+	LocalServerConnection(TCPServer_Actor& srv, RTTcpSocket& sock, bool log, long maxWait)
+			:server(srv), listenSock(sock), logToStdOut(log),
+			 hasReplied(false), maxWaitForReply(maxWait) {
+		clientSock.registerWith(&ioMonitorClient);
 	}
 
-	virtual void reply(const std::string& msg) {
-		if (hasReplied) {
-			std::cout << "Warning: Multiple reply detected for request. Message: " << msg << std::endl;
-			return;
+	~LocalServerConnection() {
+		clientSock.close();
+	}
+
+	bool init() {
+
+		if (!clientSock.acceptFrom(listenSock)) {
+			if(logToStdOut) {
+				std::cerr << "Connection failed to accept socket" << std::endl;
```

Comment:
Inconsistency. Flag checks if user wants to log to stdout, but message is written to stderr. If this is an unexpected error message, we should write to stderr without checking "logToStdOut" (as I think is already done in some other places).